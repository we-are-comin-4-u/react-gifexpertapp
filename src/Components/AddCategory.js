import React, { useState } from 'react';
import PropTypes from 'prop-types';
// import { GifExpertApp } from '../GifExpertApp'

export const AddCategory = ({setCategories}) => {

   const [inputValue,setInputValue] = useState('');

   const handleInputChanges = (e) =>{
   // console.log(e.target.value);
      setInputValue(e.target.value);
      // console.log('handleInputChangeswas called sucesfully')

   }

   const handleSubmit = (e) =>{
      e.preventDefault();
      
      // console.log('handleSubmit', inputValue)

       if(inputValue.trim().length > 2){

          setCategories(catego => [ inputValue,...catego,]);
         setInputValue('');
       }

   }

   return (
      
      <form onSubmit={handleSubmit}>
         <p>{inputValue}</p>
         <input 
         type = 'text'
         value={inputValue}
         onChange={handleInputChanges}
         />

      </form>
         
      
   )
}

AddCategory.propTypes = {
   setCategories:PropTypes.func.isRequired
}

export default AddCategory;