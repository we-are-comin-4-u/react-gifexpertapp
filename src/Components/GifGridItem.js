// import { prototype } from '@wojtekmaj/enzyme-adapter-react-17'
import React from 'react';
import PropTypes from 'prop-types';




export const GifGridItem = ({title,url}) => {

   // console.log({id,title,url});
   return (
      <div className = 'card animate__animated animate__bounce'>
         <img src = {url} alt = {title}/>
         <p>{title}</p>
      </div>
   )
}


GifGridItem.propTypes = {
   title: PropTypes.string.isRequired,
   url: PropTypes.string.isRequired,
   
   
}




// import PropTypes from 'prop-types';
