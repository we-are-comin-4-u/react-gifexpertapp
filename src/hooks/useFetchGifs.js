import { useEffect, useState } from "react"

import { getGifs } from '../helpers/getGifs';


  export const useFetchGifs = (category) => {

   const [state, setState] = useState({

      data:[],
      loading:true

   })

   

   useEffect(()=>{

      getGifs(category).then( imgs =>{

         setTimeout(()=>{

            
            setState({
               data: imgs,
               loading: false
            });

         },3000);
      })      

   },[category])

//    setTimeout(() => {
//             setState({
//                data:[1,2,3,4,5,6,7],
//                loading: false
//             })
//          },3000);

  return state;

   }