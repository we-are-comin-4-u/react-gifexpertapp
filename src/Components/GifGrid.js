import React from 'react' // { useState,useEffect }
import PropTypes from 'prop-types'

import { useFetchGifs } from '../hooks/useFetchGifs'
// import { getGifs } from '../helpers/GetGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({category}) => {

   //  const[images,setImages] = useState([])
     const {data:images,loading} = useFetchGifs(category);

   //   console.log(loading);
   //   console.log(data);
   // useEffect(()=>{
   //    getGifs(category).then(setImages);
   // },[category]);

   
      // getGifs();

   return (
      <>
         <h3 className ="animate__fadeIn">{category}</h3>

            { loading && <p className="animate__animated animate__flash">Loading...👩‍💻</p>  }

         
       <div className = "card-grid">
          {/* <h3>{count}</h3> */}
         
            
            
               {
                  images.map(img => ( 
                     <GifGridItem 
                     key = {img.id}//acostumbrate a usar la destructuracion
                        {...img} //esto es un operador expres hay mucho ejemplos en net
                     />  
                  ))

               }
               
      </div> 
      </>
   )
}

GifGrid.propTypes = {
   category: PropTypes.string.isRequired
}