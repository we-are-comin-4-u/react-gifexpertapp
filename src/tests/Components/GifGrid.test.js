import React from 'react';


import '@testing-library/jest-dom'
import { shallow } from 'enzyme'
import { GifGrid } from '../../Components/GifGrid'
import  {useFetchGifs}  from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('testing <GifGrid/>',()=>{

  
  const category = 'One Punch';
 
  
  test('should to show the snapshot',()=>{
    
    useFetchGifs.mockReturnValue({
      data:[],
      loading:true,
    });

    const wrapper = shallow(<GifGrid category ={category}/>);
    expect(wrapper).toMatchSnapshot();
    
    
  })
  
  test('should to show items whend the loadind images finished useFetchGifs', ()=>{
    
    const gifs = [{
      id: 'ABC',
      url: 'https://google.com',
      title: 'ABC Title'
  },
  {
    id: '123',
      url: 'https://google.com',
      title: 'ABC Title'
  }];
  
  
   
    
  useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
  });
  
  const wrapper = shallow(<GifGrid category={category}/>);

  expect(wrapper).toMatchSnapshot();

  expect(wrapper.find('p').exists()).toBe(false);
  expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    // const wrapper = shallow(<GifGrid category ={category}/>)


  })


})

