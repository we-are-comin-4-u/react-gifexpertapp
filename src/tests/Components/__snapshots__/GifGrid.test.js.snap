// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing <GifGrid/> should to show items whend the loadind images finished useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__fadeIn"
  >
    One Punch
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="ABC Title"
      url="https://google.com"
    />
    <GifGridItem
      id="123"
      key="123"
      title="ABC Title"
      url="https://google.com"
    />
  </div>
</Fragment>
`;

exports[`testing <GifGrid/> should to show the snapshot 1`] = `
<Fragment>
  <h3
    className="animate__fadeIn"
  >
    One Punch
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...👩‍💻
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
